libbeat/publisher/client.go

```diff
@@ -106,13 +115,9 @@ func (c *client) annotateEvent(event common.MapStr) {
 	}
 }
 
-func (c *client) getClient(opts []ClientOption) (context, eventPublisher) {
-	var ctx context
-	for _, opt := range opts {
-		ctx = opt(ctx)
-	}
-
-	if ctx.sync {
+func (c *client) getClient(opts []ClientOption) (Context, eventPublisher) {
+	ctx := makeContext(opts)
+	if ctx.Sync {
```

Comment:
The code you are looking at is in libbeat. Yes in libbeat the default is Async. It is filebeat passing the Sync flag enforing the publisher to do a blocking sync call.